Intermediate-concurrency-tutorial/src/main/java/com/vinnik/kate/TaskExecutor.java

```diff
@@ -0,0 +1,85 @@
+package com.vinnik.kate;
+
+import java.util.Random;
+
+/** This class starts threads, which are measures time of processing tasks. */
+public final class TaskExecutor {
+    private static final int UPPER_BOUND_OF_RANDOM = 10000;
+    private static final int NANOSECONDS_IN_MILLISECONDS = 1000000;
+
+    private final int amountOfThreads;
+
+    private final TaskTable taskTable;
+
+    /** Creates new object's table with tasks with given amount of tasks, creates given amount of threads. */
+    public TaskExecutor(int amountOfThreads, int amountOfMeasurements) {
+        this.amountOfThreads = amountOfThreads;
+        this.taskTable = new TaskTable(amountOfMeasurements);
+    }
+
+    /** Starts measures of time to complete each task from table with tasks. */
+    public void measureTasksCompletionTime() {
+        System.out.println("Measuring starts...");
+        for (int i = 0; i < amountOfThreads; i++) {
+            final Thread thread = new Thread(new ThreadSorter(i));
+            thread.start();
+        }
+    }
+
+    private final class ThreadSorter implements Runnable {
+        private final int threadIdentifier;
+        private Task task;
+
+        private ThreadSorter(int threadIdentifier) {
+            this.threadIdentifier = threadIdentifier;
+        }
+
+        @Override
+        public void run() {
+            while (!taskTable.isAllTaskTook()) {
+                this.task = taskTable.popLastTaskIdentifier();
+                if (this.task != null) {
```

Comment:
The scope for this one is relatively large. Isn't it simpler to check if task is null (`this.task == null`) first?